spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -155,23 +405,463 @@ class NwbSortingExtractor(se.SortingExtractor):
+                raise Exception("'default_values' list must have length equal to" +
+                                " number of features to be copied.")
+        # Copies only features that do not exist in NWB file
+        nspikes_units = self.get_nspikes()
+        for i, feat in enumerate(new_feature_names):
+            full_feat_name = 'spike_feature_' + feat
+            if full_feat_name in curr_feature_names:
+                print(full_feat_name + " already exists in NWB file and can't be copied.")
+            else:
+                feat_values = {}
+                for id in unit_ids:
+                    vals = sorting.get_unit_spike_features(unit_id=id,
+                                                           feature_name=feat)
+                    feat_values[str(id)] = vals.tolist()
+                self.set_unit_spike_features(unit_ids=unit_ids,
+                                             feature_name=feat,
+                                             values=feat_values,
+                                             default_value=default_values[i])
+
+    def clear_unit_spike_features(self, unit_id=None, feature_name=None):
+        '''NWB files do not allow removing features.'''
+        print(self.clear_unit_spike_features.__doc__)
+
+    def clear_units_spike_features(self, *, unit_ids=None, feature_name):
+        '''NWB files do not allow removing features.'''
+        print(self.clear_units_spike_features.__doc__)
+
+    def frame_to_time(self, frame):
+        return frame / self.get_sampling_frequency() + self._t0
 
-    def get_unit_spike_train(self, unit_id, start_frame=None, end_frame=None):
-        pass
+    def add_epoch(self, epoch_name, start_frame, end_frame):
+        '''This function adds an epoch to the NWB file.
+
+        Parameters
+        ----------
+        epoch_name: str
+            The name of the epoch to be added
+        start_frame: int
+            The start frame of the epoch to be added (inclusive)
+        end_frame: int
+            The end frame of the epoch to be added (exclusive)
+        '''
+        assert HAVE_NWB, "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
+        if not isinstance(epoch_name, str):
+            raise Exception("'epoch_name' must be a string")
+        if not isinstance(start_frame, int):
+            raise Exception("'start_frame' must be an integer")
+        if not isinstance(end_frame, int):
+            raise Exception("'end_frame' must be an integer")
+
+        with NWBHDF5IO(self._path, 'r+') as io:
+            nwbfile = io.read()
+            nwbfile.add_epoch(start_time=self.frame_to_time(start_frame),
+                              stop_time=self.frame_to_time(end_frame),
+                              tags=epoch_name)
+            io.write(nwbfile)
+
+    def remove_epoch(self, epoch_name=None):
+        '''NWB files do not allow removing epochs.'''
+        print(self.remove_epoch.__doc__)
```

Comment:
this common pattern begs to be replaced with some `@justprintthedoc` (or alike) decorator and a pure `pass` in the body.